symengine/tests/basic/test_parser.cpp

```diff
@@ -637,6 +638,18 @@ TEST_CASE("Parsing: doubles", "[parser]")
     s = "-.12x";
     res = parse(s);
     REQUIRE(eq(*res, *mul(real_double(-0.12), x)));
+
+    s = "2.5";
+    res = parse(s);
+    REQUIRE(eq(*res, *real_double(2.5)));
+
+#ifdef HAVE_STRTOD_L
+    std::setlocale(LC_NUMERIC, "de_DE.UTF-8");
+    s = "2.5";
+    res = parse(s);
+    REQUIRE(eq(*res, *real_double(2.5)));
+    std::setlocale(LC_NUMERIC, "C");
+#endif
```

Comment:
Why do we only run this tests when `HAVE_STRTOD_L` is defined? Is there a reason not to run this always?

Reply:
`strtod_l` is not in C standard and almost all platforms I've checked supports it. Linux, OSX, MinGW-w64 and MSVC15 supports it. Only MinGW 32-bit platform (MinGW-w64 32-bit supports it) doesn't support it.

Reply:
Yes, but what happens if the platform does not support it? It means the test will fail?

Reply:
Yes. That's why the test is only run if `HAVE_STRTOD_L` is defined.

Reply:
Hm. I would really ship our own implementation of this very important conversion, that works everywhere. Here is how musl does it:

https://github.com/ifduyue/musl/blob/79f653c6bc2881dd6855299c908a442f56cb7c2b/src/stdlib/strtod.c#L6

Reply:
We can extract it into the `symengine_strtod.h` header file, merge this PR, and then substitute `symengine_strtod.h` for our own implementation later and change this test to run everywhere.